## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 415045`
Commit: `668d78df00fa6aa8d0af2acbc312dde077211f2e`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>newsraft</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>newsraft</summary>
<pre>Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash PREFIX=\$\(out\)
cc -std=c99 -O3 `pkg-config --cflags libcurl  2>/dev/null` `pkg-config --cflags expat    2>/dev/null` `pkg-config --cflags gumbo    2>/dev/null` `pkg-config --cflags sqlite3  2>/dev/null` -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -Isrc  -o newsraft src/newsraft.c `pkg-config --libs   libcurl  2>/dev/null || echo '-lcurl'` `pkg-config --libs   expat    2>/dev/null || echo '-lexpat'` `pkg-config --libs   gumbo    2>/dev/null || echo '-lgumbo'` `pkg-config --libs   sqlite3  2>/dev/null || echo '-lsqlite3'` -lpthread
In file included from src/newsraft.c:9:
In file included from src/curses.c:3:
src/termbox2.h:2146:5: error: call to undeclared function 'cfmakeraw'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 2146 |     cfmakeraw(&tios);
      |     ^
src/termbox2.h:2300:19: error: use of undeclared identifier 'SIGWINCH'
 2300 |     if (sigaction(SIGWINCH, &sa, NULL) != 0) {
      |                   ^
src/termbox2.h:2434:15: error: use of undeclared identifier 'SIGWINCH'
 2434 |     sigaction(SIGWINCH, &sa, NULL);
      |               ^
3 errors generated.
make: *** [makefile:56: newsraft] Error 1</pre>
</details>
</details>

--- a/pkgs/by-name/ne/newsraft/package.nix
+++ b/pkgs/by-name/ne/newsraft/package.nix
@@ -6,21 +6,20 @@
   curl,
   expat,
   gumbo,
-  ncurses,
   sqlite,
   nix-update-script,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "newsraft";
-  version = "0.30";
+  version = "0.31";
 
   src = fetchFromGitea {
     domain = "codeberg.org";
     owner = "newsraft";
     repo = "newsraft";
     rev = "newsraft-${finalAttrs.version}";
-    hash = "sha256-h9gjw2EjWWNdyQT2p4wgWlz4TNitDBX5fPbNNH9/th4=";
+    hash = "sha256-XnVGt9frUKeAjxYk2cr3q3a5HpqVH0CHnNiKdTTBnqA=";
   };
 
   nativeBuildInputs = [ pkg-config ];
@@ -28,7 +27,6 @@ stdenv.mkDerivation (finalAttrs: {
     curl
     expat
     gumbo
-    ncurses
     sqlite
   ];
 
@@ -37,6 +35,8 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
     install -Dm444 doc/newsraft.desktop -t $out/share/applications
   '';
+
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-D_DARWIN_C_SOURCE";
 
   passthru.updateScript = nix-update-script { };
 

--- a/pkgs/by-name/ne/newsraft/package.nix
+++ b/pkgs/by-name/ne/newsraft/package.nix
@@ -31,10 +31,7 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   makeFlags = [ "PREFIX=$(out)" ];
-
-  postInstall = lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
-    install -Dm444 doc/newsraft.desktop -t $out/share/applications
-  '';
+  installTargets = "install install-desktop";
 
   env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-D_DARWIN_C_SOURCE";
 
@@ -42,7 +39,8 @@ stdenv.mkDerivation (finalAttrs: {
 
   meta = {
     description = "Feed reader for terminal";
-    homepage = "https://codeberg.org/grisha/newsraft";
+    homepage = "https://codeberg.org/newsraft/newsraft";
+    changelog = "https://codeberg.org/newsraft/newsraft/releases/tag/newsraft-${finalAttrs.version}";
     license = lib.licenses.isc;
     maintainers = with lib.maintainers; [
       arthsmn

